src/Tribe/Admin/Notice/Update.php

```diff
@@ -0,0 +1,177 @@
+			],
+			[ $this, 'should_display' ]
+		);
+
+		add_action( 'admin_enqueue_scripts', [ $this, 'add_block_editor_notice' ] );
+	}
+
+	/**
+	 * Should the notice be displayed?
+	 * 
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	public function should_display() {
+		$admin_helpers = tribe( 'admin.helpers' );
+
+		if ( isset( $_GET['update-message-the-events-calendar'] ) ) {
+			return false;
+		}
+
+		if ( ! $admin_helpers->is_screen() ) {
+			return false;
+		}
+
+		if ( ! $admin_helpers->is_post_type_screen() ) {
+			return false;
+		}
+
+		if ( tribe( State::class )->is_migrated() ) {
+			return false;
+		}
+
+		if ( Tribe__Admin__Notices::instance()->has_user_dismissed( $this->notice->slug ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * HTML for the notice for sites using UTC Timezones.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public function notice() {
+		ob_start();
+		?>
+		<div class="tec-update-notice">
+			<h3 class="tec-update-notice__title">
+				<?php echo esc_html( $this->update_title ); ?>
+			</h3>
+			<div class="tec-update-notice__description">
+				<?php echo esc_html( $this->update_description  ); ?>
+			</div>
+			<div class="tec-update-notice__actions">
+				<a class="tec-update-notice__button button" href="<?php echo esc_url( get_admin_url( null, $this->upgrade_tab_link ) ); ?>">
+					<?php esc_html_e( 'Migrate your site', 'the-events-calendar' ); ?>
+				</a>
+				<a class="tec-update-notice__link" href="<?php echo esc_url( $this->learn_more_link ); ?>">
+					<?php esc_html_e( 'Learn more', 'the-events-calendar' ); ?>
+				</a>
+			</div>
+		</div>
+		<?php
+		$notice = ob_get_contents();
+		ob_end_clean();
+		
+		return $notice;
+	}
+
+	/**
+	 * Add the JS for the admin notice.
+	 * 
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function add_block_editor_notice() {
+		if ( ! $this->should_display() ) {
+			return;
+		}
+
+		global $current_screen;
+		$current_screen = get_current_screen();
+
+		if ( method_exists( $current_screen, 'is_block_editor') && $current_screen->is_block_editor() ) {
+			wp_add_inline_script( 'tribe-events-editor', $this->js_notice() );
+		}
+	}
+
+	/**
+	 * Javascript for creating admin notice in the block editor.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public function js_notice() {
+		$js = '( function ( wp ) {';
+		$js .= 'const tec_update_description = "<b>' . esc_html( $this->update_title ) . '</b><p>' . esc_html( $this->update_description ) . '</p>";';
+		$js .= 'const tec_upgrade_tab_link = "' . esc_url( $this->upgrade_tab_link ) . '";';
+		$js .= 'const tec_learn_more_link = "' . esc_url( $this->learn_more_link ) . '";';
+		$js .=	"wp.data.dispatch( 'core/notices' ).createNotice('info', tec_update_description, { __unstableHTML: true, isDismissible: true, actions: [ { url: tec_upgrade_tab_link, label: 'Start storage migration' }, { url: tec_learn_more_link, label: 'Learn more' } ] } );";
+		$js .= '} )( window.wp );';
```

Comment:
Why not create a small JS file for this?

Reply:
Moved to JS file, passed data via localize param